### ISSUE TITLE
Learning with Leeper: Joystick will not go Right or Down

### ISSUE THREAD

**Issue by GreatHierophant** [AppleWin/AppleWin#1128]
Learning with Leeper requires a joystick to move but in AppleWin it will not move the character to the right or down, only up and left, which impacts gameplay from the first screen.  This game has a rather exacting joystick reading routine, it requires an analog joystick connected to my Apple IIe to get the right and down directionals working.  I use a PC to Apple joystick adapter, an analog PC joystick works but a digital one like a Gravis Gamepad does not.  

AppleWin will not move right or down even if analog-style inputs like a mouse or analog joystick are used.  I used 4am's crack of the game for both the Apple IIe and AppleWin.

**Comment by tomcw** [comment]
Confirmed that only up and left work, and right/down don't work.

Here's a link to the image:
https://mirrors.apple2.org.za/ftp.apple.asimov.net/images/educational/Learning%20with%20Leeper%20%284am%20crack%29.dsk

And here's the joystick read routine:
```
110B:A2 40      LDX #$40 
110D:8E 70 C0   STX $C070
1110:8E 7A 11   STX $117A
1113:8E 7B 11   STX $117B
1116:A2 00      LDX #$00 
1118:AD 64 C0   LDA $C064
111B:30 04      BMI $1121
111D:EA         NOP      
111E:EA         NOP      
111F:10 03      BPL $1124
1121:EE 7A 11   INC $117A
1124:AD 65 C0   LDA $C065
1127:30 04      BMI $112D
1129:EA         NOP      
112A:EA         NOP      
112B:10 03      BPL $1130
112D:EE 7B 11   INC $117B
1130:EA         NOP      
1131:CA         DEX      
1132:D0 E4      BNE $1118
1134:A2 07      LDX #$07 
1136:AD 7A 11   LDA $117A
1139:DD 69 11   CMP $1169
113C:90 14      BCC $1152
113E:DD 71 11   CMP $1171
1141:B0 0F      BCS $1152
1143:AD 7B 11   LDA $117B
1146:DD 59 11   CMP $1159
1149:90 07      BCC $1152
114B:DD 61 11   CMP $1161
114E:B0 02      BCS $1152
1150:90 03      BCC $1155
1152:CA         DEX      
1153:10 E1      BPL $1136
1155:8E 79 11   STX $1179
1158:60         RTS      
```

**Comment by NRS-NewRisingSun** [comment]
For this joystick routine, in AppleWin, full left-center-full right is: 40-6B-9E, failing to exceed the threshold for right, at A0. What I don't understand is whe X-Trim and Y-Trim settings in the user interface do not seem to change the read result.

**Comment by tomcw** [comment]
A fix is to change this line's value from `280` to `287`:
(A value of `286` worked for right, but not down.)

https://github.com/AppleWin/AppleWin/blob/f300edb5dee9479fbf9ebdb289b4a69dc6ca6440/source/Joystick.cpp#L668-L670

**Comment by tomcw** [comment]
Fix in this new [AppleWin 1.30.12.0](https://github.com/AppleWin/AppleWin/releases/tag/v1.30.12.0).
